testing/unittests/loggingtests.cpp

```diff
@@ -0,0 +1,485 @@
+        hasService("hasService-none/update", agent, LGSTUpdateLOG, false);
+        hasService("hasService-none/id", agent, LGSTGetTransactionID, false);
+    }
+    void testHasService_Seed()
+    {
+        const char* agentText = R"!!!(
+            <LogAgent name="mock" module="mock">
+              <GetTransactionSeed seed-id="12345" status-code="2" status-message="status 2"/>
+            </LogAgent>
+        )!!!";
+        Owned<IEspLogAgent> agent(createAgent(agentText));
+
+        hasService("hasService-seed/seed", agent, LGSTGetTransactionSeed, true);
+        hasService("hasService-seed/update", agent, LGSTUpdateLOG, false);
+        hasService("hasService-seed/id", agent, LGSTGetTransactionID, false);
+    }
+    void testHasService_Update()
+    {
+        const char* agentText = R"!!!(
+            <LogAgent name="mock" module="mock">
+              <UpdateLog response="update response" status-code="1" status-message="status 1"/>
+            </LogAgent>
+        )!!!";
+        Owned<IEspLogAgent> agent(createAgent(agentText));
+
+        hasService("hasService-update/seed", agent, LGSTGetTransactionSeed, false);
+        hasService("hasService-update/update", agent, LGSTUpdateLOG, true);
+        hasService("hasService-update/id", agent, LGSTGetTransactionID, false);
+    }
+    void testHasService_Id()
+    {
+        const char* agentText = R"!!!(
+            <LogAgent name="mock" module="mock">
+              <GetTransactionId id="12345T67890"/>
+            </LogAgent>
+        )!!!";
+        Owned<IEspLogAgent> agent(createAgent(agentText));
+
+        hasService("hasService-id/seed", agent, LGSTGetTransactionSeed, false);
+        hasService("hasService-id/update", agent, LGSTUpdateLOG, false);
+        hasService("hasService-id/id", agent, LGSTGetTransactionID, true);
+    }
+    void testHasService_All()
+    {
+        const char* agentText = R"!!!(
+            <LogAgent name="mock" module="mock">
+              <GetTransactionSeed seed-id="12345" status-code="2" status-message="status 2"/>
+              <UpdateLog response="update response" status-code="1" status-message="status 1"/>
+              <GetTransactionId id="12345T67890"/>
+            </LogAgent>
+        )!!!";
+        Owned<IEspLogAgent> agent(createAgent(agentText));
+
+        hasService("hasService-all/seed", agent, LGSTGetTransactionSeed, true);
+        hasService("hasService-all/update", agent, LGSTUpdateLOG, true);
+        hasService("hasService-all/id", agent, LGSTGetTransactionID, true);
+    }
+
+    void testGetTransactionSeed_True()
+    {
+        const char* agentText = R"!!!(
+            <LogAgent name="mock" module="mock">
+              <GetTransactionSeed seed-id="67890" status-code="0" status-message="status 0"/>
+            </LogAgent>
+        )!!!";
+        Owned<IEspLogAgent> agent(createAgent(agentText));
+        
+        getTransactionSeed("getTransactionSeed-true", agent, true, "67890", 0, "status 0");
+    }
```

Comment:
The mock object is being tested extensively because my choice was to mock a complete log agent plugin for the log manager being tested elsewhere can load, or take advantage of the modular log agent by creating a module in the existing plugin. I chose the latter and a mock object accessible in production should be well tested.

An agent module seemed like it would require less code, would further demonstrate the flexibility of the modular log agent framework, and I couldn't rule out the potential usefulness of being able to simulate agent responses.

Reply:
This test test suite is testing the mock agent, for previously stated reasons. Some tests have been removed to simplify the suite.